nemo/collections/nlp/modules/common/decoder_module.py

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Dict, Optional
+
+from nemo.core.classes import NeuralModule
+from nemo.core.neural_types import ChannelType, MaskType, NeuralType
+
+__all__ = ['DecoderModule']
+
+
+class DecoderModule(NeuralModule):
+    @property
+    def input_types(self) -> Optional[Dict[str, NeuralType]]:
+        return {
+            "input_ids": NeuralType(('B', 'T'), ChannelType()),
+            "decoder_mask": NeuralType(('B', 'T'), MaskType(), optional=True),
+            "encoder_mask": NeuralType(('B', 'T'), MaskType(), optional=True),
+            "encoder_hidden_states": NeuralType(('B', 'T', 'D'), ChannelType()),
```

Comment:
this should be optional as well. A language model is decoder-only without encoder_hidden_states

Reply:
Done.